<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>visualCornerSelect</class>
 <widget class="QWidget" name="visualCornerSelect">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>786</width>
    <height>359</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>visualCornerSelect</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QFrame" name="frame_2">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="ImageLabel" name="image_label_original">
        <property name="text">
         <string>Image</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="ImageLabel" name="image_label_no_perspective">
        <property name="text">
         <string>Image</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item alignment="Qt::AlignHCenter">
    <widget class="QGroupBox" name="groupBox_selects">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Detection algoritms</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="DescriptiveComboBox" name="corner_detector_select" native="true"/>
      </item>
      <item row="0" column="1">
       <widget class="DescriptiveComboBox" name="inventory_detector_select" native="true"/>
      </item>
      <item row="0" column="2">
       <widget class="QCheckBox" name="checkBox_show_borders">
        <property name="text">
         <string>Show borders</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QPushButton" name="pushButton_set_corners">
        <property name="text">
         <string>Set Corners</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QCheckBox" name="checkBox_show_grid">
        <property name="text">
         <string>Show grid</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_click_corners">
        <property name="styleSheet">
         <string notr="true">color: rgb(46, 194, 126);</string>
        </property>
        <property name="text">
         <string>Now click 4 corners on left image</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QCheckBox" name="checkBox_show_inventories">
        <property name="text">
         <string>Show inventories</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ImageLabel</class>
   <extends>QLabel</extends>
   <header>GUI/widgets/ImageLabel.h</header>
   <slots>
    <signal>clicked()</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>DescriptiveComboBox</class>
   <extends>QWidget</extends>
   <header>GUI/widgets/DescriptiveComboBox.h</header>
   <container>1</container>
   <slots>
    <signal>element_changed(QVariant)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>image_label_original</sender>
   <signal>clicked()</signal>
   <receiver>visualCornerSelect</receiver>
   <slot>on_image_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>110</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>134</x>
     <y>-13</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_set_corners</sender>
   <signal>clicked()</signal>
   <receiver>visualCornerSelect</receiver>
   <slot>on_set_corners_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>412</x>
     <y>308</y>
    </hint>
    <hint type="destinationlabel">
     <x>551</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>corner_detector_select</sender>
   <signal>element_changed(QVariant)</signal>
   <receiver>visualCornerSelect</receiver>
   <slot>on_corner_detector_changed(QVariant)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>258</x>
     <y>273</y>
    </hint>
    <hint type="destinationlabel">
     <x>-3</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>inventory_detector_select</sender>
   <signal>element_changed(QVariant)</signal>
   <receiver>visualCornerSelect</receiver>
   <slot>on_inventory_detector_changed(QVariant)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>429</x>
     <y>273</y>
    </hint>
    <hint type="destinationlabel">
     <x>-2</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_show_borders</sender>
   <signal>clicked(bool)</signal>
   <receiver>visualCornerSelect</receiver>
   <slot>on_show_borders_switched(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>470</x>
     <y>259</y>
    </hint>
    <hint type="destinationlabel">
     <x>672</x>
     <y>258</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_show_grid</sender>
   <signal>clicked(bool)</signal>
   <receiver>visualCornerSelect</receiver>
   <slot>on_show_grid_switched(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>486</x>
     <y>285</y>
    </hint>
    <hint type="destinationlabel">
     <x>650</x>
     <y>280</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_show_inventories</sender>
   <signal>clicked(bool)</signal>
   <receiver>visualCornerSelect</receiver>
   <slot>on_show_inventories_switched(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>538</x>
     <y>319</y>
    </hint>
    <hint type="destinationlabel">
     <x>666</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>corner_detector_changed(object)</signal>
  <signal>inventory_detector_changed(object)</signal>
  <slot>on_corner_detector_changed(QVariant)</slot>
  <slot>on_inventory_detector_changed(QVariant)</slot>
  <slot>on_image_clicked()</slot>
  <slot>on_set_corners_clicked()</slot>
  <slot>on_show_borders_switched(bool)</slot>
  <slot>on_show_grid_switched(bool)</slot>
  <slot>on_show_inventories_switched(bool)</slot>
 </slots>
</ui>
